proof/invariant-abstract/TcbAcc_AI.thy

```diff
@@ -2540,8 +2532,8 @@ lemma thread_set_domain_fault_tcbs_valid_states[wp]:
   "thread_set_domain t d \<lbrace> fault_tcbs_valid_states \<rbrace>"
   by (simp add: thread_set_domain_def thread_set_fault_tcbs_valid_states_trivial)
 
-lemma thread_set_domain_invs[wp]:
-  "thread_set_domain t d \<lbrace>invs\<rbrace>"
+lemma (in TcbAcc_AI) thread_set_domain_invs[wp]:
+  "thread_set_domain t d \<lbrace>invs :: 'state_ext state \<Rightarrow> _\<rbrace>"
```

Comment:
can you clarify why the state type annotations are now needed? I'm seeing a fair few of them

Reply:
The commits being merged in from master moved `thread_set_valid_arch_state` into the `TcbAcc_AI` locale and changed it to use the fixed `state_ext` type from it, which meant that these `thread_set_domain` and `thread_set_priority` lemmas needed it as well.